### DISCUSSION TITLE
Windows 7 / Server 2008R2 support is marching toward its end in Mesa3D

### DISCUSSION THREAD

**Discussion by pal1000** [pal1000/mesa-dist-win#112] [Polls]
First step was done in [Mesa3D 22.2 development cycle](https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/17431) where [Futex API](https://en.wikipedia.org/wiki/Futex) is used by default bumping minimum requirements to Windows 8 / Server 2012. It can still be disabled for now so this poll purpose is to decide how to handle this for Mesa3D 22.2 stable cycle and beyond.

**Comment by Directox01** [comment]
I believe if we stop supporting win7 and 2008R2 you'll have much more time to focus on the newest version. So I believe there is no reason to waste time on win7 and 2008. 
My vote: is NO

**Reply by pal1000** [reply]
Please read https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/17122 for consequences of keeping Windows 7 / Server 2008R2 support. It seams there is some performance penalty involved.  I don't know if keeping Windows 7 / Server 2008R2 affects performance while running on Windows 8 / Server 2012 or newer.

**Reply by pal1000** [reply]
Keeping Windows 7 / Server 2008R2 support [bit us as you predicted](https://github.com/pal1000/mesa-dist-win/issues/118).

**Comment by Svyatpro** [comment]
I even want to mention it would be highly appreciated if there was a build for XP/2003

**Reply by pal1000** [reply]
Support for XP/Server 2003 was dropped a really long time ago and honestly I don't recall when.

**Reply by Svyatpro** [reply]
to be honest your latest opengl32.dll software wrapper works for me on XP/2003 with some compatibility hacks. But vulkan software icd doesn't work properly and I don't even know the reason. But it doesn't even work on Win10.
It is being detected as llvmpipe (LLVM 15.0.5, 256 bits) by vulkan-info but no one Vulcan demo works and also dxvk crashes with it

**Reply by pal1000** [reply]
Vulkan was never supported on XP.
I don't know much about lavapipe crashing everything problem.  Collecting a MSVC call stack or MinGW backtrace may help, You can make use of  MSVC debug  info and MinGW debug binaries.

**Reply by LionsPhil** [reply]
(An old release note claims "remove a relic about Windows XP/PosReady 2009, it is no longer supported since Mesa3D 17.2.0", so I assume [17.1.8.401-1](https://github.com/pal1000/mesa-dist-win/releases/tag/17.1.8.401-1) is probably the last to intentionally support XP?)

**Reply by TNT-333** [reply]
it works with one core api

**Comment by Tarek-Hasan** [comment]
Older windows support will help people avoid M$ force upgrade, enshittification and AI nonsense for longer period of time. But why 32bit support!?! It's nothing but technological baggage. I used 64bit windows on my 20 years old Pentium 4 PC until its death. Using 32bit OS is nothing but pure boomer stubbornness.

**Reply by NS-Clone** [reply]
blahblahblah
10 years ago most software on your 10 years old pentium 4 PC was x32
and NOT only all software updating every day to support newer not compatible M$ things

**Reply by Tarek-Hasan** [reply]
@NS-Clone, you're just a troll who promotes 32bit system for God knows what reason. Just another "x86 purist" retard. You constantly harass devs for 32 version of software only 2 or 3 people like you will use.

<img width="977" height="400" alt="Screenshot 2026-02-19 113030" src="https://github.com/user-attachments/assets/adc50b23-1b81-4a36-ad47-3f2b3427c65f" />

<img width="985" height="395" alt="Screenshot 2026-02-19 113942" src="https://github.com/user-attachments/assets/a6ee6694-0936-497d-bd19-bb5b795ccaa1" />

<img width="972" height="483" alt="Screenshot 2026-02-19 114604" src="https://github.com/user-attachments/assets/368f0fad-ec50-4f46-b7a2-f111476c7c2c" />

<img width="973" height="500" alt="Screenshot 2026-02-19 114519" src="https://github.com/user-attachments/assets/e68ba913-6069-4fe8-9942-0827a2cc61f5" />

**Reply by NS-Clone** [reply]
nope it's you just troll with that "blahblahblah i want only 64 bit version even if 32 works well"
i'm still using 32 bit systems 
also you forgot to mention that "security updates"

> Older windows support will help people avoid M$ force upgrade

also M$ force YOU to upgrade
by prohibiting 32bit drivers in 2015
so software renderers mostly needs on 32bit that not have any video drivers at all for last 10 years
than on this x64 (now even 3080 drivers can works on x64 vista)

also you have thousands x32 only software that NEVER be updated
and one day M$ just prohibit to make wow64 userspace video drivers
so you will get x64 only 3d

> only 2 or 3 people like you will use.

only 2 or 3 peoples using software renderers

and actually you NOT need software renderer on your x64 win11
and it's not usable for never x64 games
due to low performance

if you not of this 2 or 3 peoples
go to download security updates for your win11

some things can't be done without x64
if you not need more than 2GB per process 
x32 is enough
and will works on both
until M$ completely cancel x32

> You constantly harass devs 

why you harass devs with your x64 only?
also trolls here usually have hidden activity

**Reply by Tarek-Hasan** [reply]
@NS-Clone, I'm not sure if you are a retarded "32bit purist" or a CIA/Mossad agent who wants every software vulnerable with legacy architecture supported enabled.

As I mentioned above, I've no problem with people using older hardware or older software. But if you want latest version support of a software, use at least a 64bit version OS or use an older version. Don't drag down the tech world with you into a quicksand.

Older architecture support prevents dev from implementing optimization for newer architectures, spend time and other resources into fixing bugs for handful of "tech elites" (read - fat nerd) and ultimately make the software vulnerable for attack.

You may be mentally obsessed with 32bit architecture. Go see a psychiatrist.

**Reply by NS-Clone** [reply]
also in mentioned issues
3 from 4 projects already have x86 x32 builds itself
so you even doesn't understand real issue

voidImageViewer itself works on x32 XP but builded with msvc
i just build it with gcc and mention patches to do it
and also add avif suport

so it's you troll not me
previous post only proof of this

**Reply by NS-Clone** [reply]
> I'm not sure if you are a retarded "32bit purist" or a CIA/Mossad agent who wants every software vulnerable with legacy architecture supported enabled.

>  and one day M$ just prohibit to make wow64 userspace video drivers
so you will get x64 only 3d

also found that nvidia already dropped x32 for opencl and cuda in latest x64 drivers 👏 
and how you will run something old and x32 that request it? :rofl:
...and most of prebuilt software opencl-s are... 
yep it's x64
yep maybe it's pretty rare software...
but same will be with d3d, opengl and vulkan soon

it's funny 15 years ago n00bs shouted "why you can't just use your XP in VM on w10?"
now XP dropped even in all VM-s (guest additions 3d acceleration)
reason as always "blahblahblah vulnerable blahblahblah security updates"
